stakingModel/migrations/3_mainnet.js

```diff
@@ -150,6 +152,33 @@ module.exports = async function(deployer, network) {
   console.log("starting...");
   await Promise.all([reserve.start(), fundManager.start()]);
 
+  console.log("deploying fund manager setReserve scheme...");
```

Comment:
why aren't we passing reserve in the constructor?

Reply:
because we set the fund manager address in the reserve constructor, we have to deploy the fund manager first. only after the reserve has deployed we can set its address in the fund manager